More TypeScript

robojumper opened the pull request:
A rebase of #140 with more typing for more components and Logic.

cjs8487 committed:
images to ts

cjs8487 committed:
item -> ts

cjs8487 committed:
items -> ts

cjs8487 committed:
blocks -> ts

cjs8487 committed:
grid tracker -> ts

cjs8487 committed:
item tracker -> ts

cjs8487 committed:
fix scaling in quest items

cjs8487 committed:
migrate some misc components

cjs8487 committed:
permalink support -> ts

robojumper committed:
Logic helpers, individual components -> ts

robojumper committed:
More components

robojumper committed:
CubeTracker -> ts

robojumper committed:
LocationGroup -> ts

robojumper committed:
LocationGroupHeader/ContextMenu -> ts

robojumper committed:
Customization -> ts

robojumper committed:
Small lint fixes

robojumper committed:
Most of Logic -> ts

robojumper committed:
Some fixups

robojumper committed:
LocationTracker -> ts

robojumper committed:
ImageLink -> ts

robojumper committed:
Tracker -> ts

robojumper committed:
More fixes

robojumper committed:
Revert some changes from the original PR

robojumper committed:
More type safety with lints

robojumper committed:
Types for options

robojumper committed:
Fix one reactivity issue

robojumper committed:
rm service worker, fix test running

cjs8487 commented on src/logic/LogicHelper.tsx:
```diff
@@ -1,16 +1,36 @@
+
+interface EvaluatedBooleanExpression {
+    items: EvaluatedRequirement[];
+    type: 'and' | 'or';
```
This should use the defined type from BooleanExpression

cjs8487 reviewed (changes requested):
Overall looks pretty solid. Some notes
- File naming schemes
 - A file should only use the `tsx` extension when it uses JSX. There's only a few places where this isn't the case, notably `OptionData`
 - We use `UpperCamelCase` for everything that isn't auotgenerated, so `react-utils` is named wrong.
- Feel free to omit unused variables from returned tuples, no need to use the underscore notation
- I see two added dependencies not referenced anywhere in the changed code, are they intentional?
- In case of boolean expressions, I'm inclined to prefer an enum over a union type, since the actual for of the operation isn't relevant, only the enumerated value. It makes APIs more stable so we're not comparing to hardcoded string values
- In the logic subsystem, where possible/reasonable, I'd like to move away from using object destructuring in the parameters. It's a leftover from what this system was based on, and is largely clunky. If it's unreasonable to make them individual parameters, we should define a type for the parameter object, rather than defining the type inline.
- For KeyDownWrapper, you should be able to find the type based on where it's used (check the prop it's passed into). It should pretty much be a standard event handler

robojumper committed:
Naming, deps

robojumper committed:
Less destructuring in logic params

robojumper committed:
Fix keyDownWrapper types

robojumper commented:
Good points - I had refrained from making too many changes in the actual code until the types were complete, but I think I have addressed your feedback.

robojumper committed:
A bit more reuse

robojumper committed:
One less accidental tsx extension
